src/dataloader.py

```diff
@@ -59,3 +63,29 @@ def __init__(self, dataset1, dataset2, *args, **kwargs):
 
     def __iter__(self):
         return zip(self.dataloader1, self.dataloader2)
+
+
+class NewVerticalDataLoader:
```

Comment:
With this dataloader, I think we can remove `PartitionDistributingDataLoader`.

To clarify roles, I would rename the new dataloader to `VerticalDataLoader` and rename the existing `VerticalDataLoader` to `SinglePartitionDataLoader`.

Reply:
I agree. Maybe we should Merge the PR for now to avoid more conflicts with the PSI PR #23? And then change these in a future PR?

Reply:
Okay